### PR TITLE
New serverless pattern - cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet

### PR DESCRIPTION
**Pull request by Gryczka**
*Issue #, if available:* #2243

*Description of changes:* Adding a new serverless pattern for a CloudFront Distribution with a CloudFront Function for Routing to Different Geo-Localized Content in an S3 Statically Hosted Website

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/example-pattern.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{
+    "title": "CloudFront to CloudFront Functions with URI Rewrite to S3",
+    "description": "Create a Static Website fronted by CloudFront with CloudFront Functions automatically routing visitors to localized content using CloudFront's Geo Location Headers.",
+    "language": ".NET",
+    "level": "200",
+    "framework": "CDK",
+    "introBox": {
+      "headline": "How it works",
+      "text": [
+        "This pattern shows how to create a CloudFront Distribution with a CloudFront Function for Routing to Different Geo-Localized Content in an S3 Statically Hosted Website. Implemented in AWS CDK .NET.",
```
Comment (bfreiberg) [inline cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/example-pattern.json:10]:
```suggestion
        "This pattern shows how to create a CloudFront Distribution with a CloudFront Function for routing to different geo-localized content in an S3 statically hosted website. Implemented in AWS CDK .NET.",
```

File: cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/example-pattern.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        "CloudFront Distributions can be configured to forward the 'CloudFront-Viewer-Country' header.  CloudFront Functions allow for javascript code to execute at CloudFront edge locations, and allow you to create url routing and rewrite rules based on request headers.  In this pattern we use the "CloudFront-Viewer-Country" header to determine the location of the request and rewrite the URI to point to different content based on the location of the request.  Specifically in this example that means pointing to an index.html in a prefix that is applied based on the location.",
```
Comment (bfreiberg) [inline cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/example-pattern.json:11]:
```suggestion
        "CloudFront Distributions can be configured to forward the 'CloudFront-Viewer-Country' header.  CloudFront Functions allow for Javascript code to execute at CloudFront edge locations, and allow you to create URL routing and rewrite rules based on request headers.  In this pattern we use the 'CloudFront-Viewer-Country' header to determine the location of the request and rewrite the URI to point to different content based on the location of the request.  Specifically in this example that means pointing to an index.html in a prefix that is applied based on the location.",
```

File: cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    ```
+1. Other useful commands:
+    ```
+    cdk diff         compare deployed stack with current state    
+    cdk synth        emits the synthesized CloudFormation template
+    ```
+
+## How it works
+
+This pattern shows how to create a CloudFront Distribution with a CloudFront Function for Routing to Different Geo-Localized Content in an S3 Statically Hosted Website. Implemented in AWS CDK .NET.
```
[the remark this answers is not in the document]
Comment (Gryczka) [reply cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/README.md:47]: It is.  Glad to remove.

**Comment by Gryczka** [comment]
@bfreiberg Let me know if there's anything else to address.

File: cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/README.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
+# CloudFront Distribution with a CloudFront Function for URI Rewrite to Different Geo-Localized Content in an S3 Statically Hosted Website using AWS CDK .NET
```
Comment (bfreiberg) [inline cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/README.md:1]:
```suggestion
# Amazon CloudFront Distribution with a CloudFront Function for URI Rewrite to Different Geo-Localized Content in an Amazon S3 Statically Hosted Website using AWS CDK .NET
```

File: cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/example-pattern.json [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      "text": [
+        "Delete the stack: <code>cdk destroy</code>."
+      ]
+    },
+    "authors": [
+      {
+        "name": "Patrick Gryczka",
+        "image": "https://www.gryczka.net/img/patryk.jpg",
+        "bio": "Patrick Gryczka is a Senior Solutions Architect at AWS specializing in Serverless technologies and the Sports industry.",
+        "linkedin": "https://www.linkedin.com/in/gryczka/"
```
Comment (bfreiberg) [inline cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/example-pattern.json:55]:
```suggestion
        "linkedin": "gryczka"
```

File: cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/example-pattern.json [hunk 6]
```diff
@@ -0,0 +1,58 @@
+{
+    "title": "CloudFront to CloudFront Functions with URI Rewrite to S3",
+    "description": "Create a Static Website fronted by CloudFront with CloudFront Functions automatically routing visitors to localized content using CloudFront's Geo Location Headers.",
```
Comment (bfreiberg) [inline cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/example-pattern.json:3]: The description has a maximum length of 150 characters, please shorten it to fit within this limit

**Review by bfreiberg — COMMENTED** [review]
Looks good overall, just a few comments left

File: cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/example-pattern.json (hunk 6 above)
Comment (Gryczka) [reply cloudfront-cff-s3-geo-uri-rewrite-cdk-dotnet/example-pattern.json]: Updated.

**Comment by Gryczka** [comment]
Thanks @bfreiberg accepted the changes and reduced the description to 150 characters.

**Comment by bfreiberg** [comment]
Looks good, thanks for your contribution. A Developer Advocate will merge the changes and publish the pattern to [serverlessland.com](http://serverlessland.com/).

**Pull request merged**